src/linear_samplers/block_col_rand_cyclic.jl

```diff
@@ -0,0 +1,92 @@
+# This code was written by Nathaniel Pritchard
+"""
+    LinSysBlkColRandCyclic <: LinSysBlkColSampler 
+
+A mutable structure with fields to handle randomly permuted block sampling. Can allow for fixed
+blocks or blocks whose entries are randomly permuted. After each cycle, a new random ordering is
+created. If the last block would be smaller than the others, columns from a previous block are 
+added to keep all blocks the same size.
+
+# Fields
+- `blockSize::Int64` - Specifies the size of each block.
+- `constantBlock::Bool` - A variable specifying if the user would like for the columns to be randomly
+permuted. 
+- `nBlocks::Int64` - Variable that contains the number of blocks overall.
+- `order::Vector{Int64}` - The order that the blocks will be used to generate updates.
+- `blocks::Vector{Int64}` - The list of all the columns in each block.
+
+Calling `LinSysBlkColRandCyclic()` defaults to setting `blockSize` to 2 and `constantBlock` to true. The `sample`
+function will handle the re-initialization of the fields once the system is provided.
+"""
+mutable struct LinSysBlkColRandCyclic <: LinSysBlkColSampler 
+    blockSize::Int64
+    constantBlock::Bool
+    nBlocks::Int64
+    order::Vector{Int64}
+    blocks::Vector{Int64}
+end
+
+LinSysBlkColRandCyclic(blockSize, constantBlock) = LinSysBlkColRandCyclic(blockSize, constantBlock, 1, Int64[], Int64[])
+LinSysBlkColRandCyclic(blockSize) = LinSysBlkColRandCyclic(blockSize, true, 1, Int64[], Int64[])
+LinSysBlkColRandCyclic() = LinSysBlkColRandCyclic(2, true, 1,  Int64[], Int64[])
+
+# Common sample interface for linear systems
+function sample(
+    type::LinSysBlkColRandCyclic,
+    A::AbstractArray,
+    b::AbstractVector,
+    x::AbstractVector,
+    iter::Int64
+)
+    if iter == 1
+        m, n = size(A)
+        # Determine number of blocks
+        type.nBlocks = div(n, type.blockSize) + (rem(n, type.blockSize) == 0 ? 0 : 1)
```

Comment:
How does this specify the number of blocks? It is setting it to either 0 or 1, correct?

Reply:
Ah sorry, I did not see the parentheses!